File: 0001-0500/0009/0009.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution
{
public:
    bool isPalindrome(int x)
    {
        if (x < 0 || (x % 10 == 0 && x != 0))
            return false;
        int res = 0;
        while (res < x)
        {
            res = res * 10 + x % 10;
            x /= 10;
        }

        return x == res || x == res / 10;
    }
};
// @lc code=end

File: 0001-0500/0009/0009.palindrome-number.js
/*
 * @lc app=leetcode id=9 lang=javascript
 *
 * [9] Palindrome Number
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
    if (x < 0 || (x !== 0 && x % 10 === 0)) {
        return false
    }
    else {
        var res = 0
        var temp = x
        while (temp > 0) {
            res = res * 10 + temp % 10
            temp = Math.floor(temp / 10)
        }
        if (res === x) {
            return true
        }
        return false
    }
};
// @lc code=end

